canlink/writer/writer.go

```diff
@@ -0,0 +1,68 @@
+package writer
+
+import (
+	"os"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// Responsible for managing the trace file
+type Writer struct {
+	traceFile       *os.File
+	fileExtention   string
+	convertToString func(*zap.Logger, *TimestampedFrame) string
+	l               *zap.Logger
+}
+
+func NewWriter(l *zap.Logger, fileExtention string) *Writer {
+	convertToStringMapping := map[string]func(*zap.Logger, *TimestampedFrame) string{
+		".jsonl": convertToJson,
+		".asc":   convertToAscii,
+	}
+	return &Writer{
+		l:               l,
+		fileExtention:   fileExtention,
+		convertToString: convertToStringMapping[fileExtention],
+		traceFile:       nil,
+	}
+}
+
+func (w *Writer) CreateTraceFile(traceDir string, busName string) error {
+	file, err := createEmptyTraceFile(traceDir, busName, w.fileExtention)
+	w.traceFile = file
+	if err != nil {
+		w.l.Info("cannot create trace file")
+		return errors.Wrap(err, "creating trace file")
+	}
+
+	return nil
+}
+
+func (w *Writer) WriteFrameToFile(frame *TimestampedFrame) error {
+	line := w.convertToString(w.l, frame)
+
+	_, err := w.traceFile.WriteString(line)
+	if err != nil {
+		w.l.Info("cannot write to file")
+		return errors.Wrap(err, "writing to trace file")
+	}
+
+	_, err = w.traceFile.WriteString("\n")
```

Comment:
You could expect the `.convertToString` function to include the `\n` character. I think that makes more sense than adding it here.